### ISSUE TITLE
When using minstall with local modules folder npm shrinkwrap does not work

### ISSUE THREAD

**Issue by stheuer** [heikomat/minstall#23]
When using minstall with local modules folder npm shrinkwrap does not work.

Reproduction:

run npm shrinkwrap after following the example in the README.md or any other project that use local modules folder leads to

λ npm shrinkwrap
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "shrinkwrap"
npm ERR! node v4.4.4
npm ERR! npm  v2.15.8

npm ERR! Problems were encountered
npm ERR! Please correct and try again.
npm ERR! extraneous: test_module@1.0.0 C:\workspace\playgound\minstal_schrinkwrap\node_modules\test_module
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     C:\workspace\playgound\minstal_schrinkwrap\npm-debug.log

**Comment by heikomat** [comment]
**TL;DR**: run `npm prune`, then `npm shrinkwrap` and then `npm install` again

@stheuer Thanks for the info! I haven't used minstall in combination with shrinkwrap before. Your error says that it doesn't shrinkwrap because it finds extraneous packages, a.k.a. you have packages in your node_modules that are not in your main package.json.

This is no surprise, because the dependencies of the submodules are, through minstall, also installed there, and your local modules are linked there aswell. according to https://github.com/npm/npm/issues/4435 you can try to run `npm prune` before you run `npm shrinkwrap`.

`npm prune` will remove all installed packages that are not in your main package.json. This will allow you to run `npm shrinkwrap`, but because some dependencies for the submodules are now missing, you have to run `npm install` again before working with your project again.

**Comment by stheuer** [comment]
I used the following setup

my-modular-app
├── modules
│   ├── test_module
│       ├── index.js  
│       └──  package.json  
├── index.js
└── package.json []

with
my-modular-app/package.json
```
{
  "name": "minstal_schrinkwrap",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "postinstall": "minstall"
  },
  "author": "Stefan Heuer <stefan.heuer@huf-group.com>",
  "license": "ISC",
  "dependencies": {
    "minstall": "1.6.1",
    "winston": "2.3.1"
  }
}
```
my-modular-app/modules/package.json
```
{
  "name": "test_module",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "Stefan Heuer <stefan.heuer@huf-group.com>",
  "license": "ISC"
}
```

**Comment by heikomat** [comment]
@stheuer as i said, can you please run `npm prune`, then `npm shrinkwrap` and then `npm install` again, and tell me if this worked?

**Comment by stheuer** [comment]
Seems to work fine

**Comment by heikomat** [comment]
ok, then i'll close this issue :)
